// magic-hour/dist/resources/v1/ai-photo-editor/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiPhotoEditorClient = void 0;
const core_1 = require("../../../core");
const post_v1_ai_photo_editor_body_1 = require("../../../types/post-v1-ai-photo-editor-body");
const post_v1_ai_photo_editor_response_1 = require("../../../types/post-v1-ai-photo-editor-response");
class AiPhotoEditorClient extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
    }
    /**
     * AI Photo Editor
     *
     * > **NOTE**: this API is still in early development stages, and should be avoided. Please reach out to us if you're interested in this API.
     *
     * Edit photo using AI. Each photo costs 10 frames.
     *
     * POST /v1/ai-photo-editor
     */
    create(request, opts) {
        return this._client.makeRequest({
            method: "post",
            path: "/v1/ai-photo-editor",
            auth: ["bearerAuth"],
            contentType: "application/json",
            body: post_v1_ai_photo_editor_body_1.Schemas$PostV1AiPhotoEditorBody.out.parse(request),
            responseSchema: post_v1_ai_photo_editor_response_1.Schemas$PostV1AiPhotoEditorResponse.in,
            opts,
        });
    }
}
exports.AiPhotoEditorClient = AiPhotoEditorClient;
