// magic-hour/dist/types/post-v1-ai-clothes-changer-body-assets.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schemas$PostV1AiClothesChangerBodyAssets = void 0;
const core_1 = require("../core");
const z = __importStar(require("zod"));
/**
 * Takes network data, validates it, and transforms keys to match typescript object PostV1AiClothesChangerBodyAssets
 */
const SchemaIn$PostV1AiClothesChangerBodyAssets = z
    .object({
    garment_file_path: z.string(),
    garment_type: z.enum(["dresses", "lower_body", "upper_body"]),
    person_file_path: z.string(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        garment_file_path: "garmentFilePath",
        garment_type: "garmentType",
        person_file_path: "personFilePath",
    });
});
/**
 * @internal
 * Takes typescript data, validates it, and maps keys to match the expected external object External$PostV1AiClothesChangerBodyAssets
 */
const SchemaOut$PostV1AiClothesChangerBodyAssets = z
    .object({
    garmentFilePath: z.string(),
    garmentType: z.enum(["dresses", "lower_body", "upper_body"]),
    personFilePath: z.string(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        garmentFilePath: "garment_file_path",
        garmentType: "garment_type",
        personFilePath: "person_file_path",
    });
});
exports.Schemas$PostV1AiClothesChangerBodyAssets = {
    in: SchemaIn$PostV1AiClothesChangerBodyAssets,
    out: SchemaOut$PostV1AiClothesChangerBodyAssets,
};
