// magic-hour/dist/types/post-v1-face-swap-body-assets.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schemas$PostV1FaceSwapBodyAssets = void 0;
const core_1 = require("../core");
const z = __importStar(require("zod"));
/**
 * Takes network data, validates it, and transforms keys to match typescript object PostV1FaceSwapBodyAssets
 */
const SchemaIn$PostV1FaceSwapBodyAssets = z
    .object({
    image_file_path: z.string(),
    video_file_path: z.string().optional(),
    video_source: z.enum(["file", "youtube"]),
    youtube_url: z.string().optional(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        image_file_path: "imageFilePath",
        video_file_path: "videoFilePath",
        video_source: "videoSource",
        youtube_url: "youtubeUrl",
    });
});
/**
 * @internal
 * Takes typescript data, validates it, and maps keys to match the expected external object External$PostV1FaceSwapBodyAssets
 */
const SchemaOut$PostV1FaceSwapBodyAssets = z
    .object({
    imageFilePath: z.string(),
    videoFilePath: z.string().optional(),
    videoSource: z.enum(["file", "youtube"]),
    youtubeUrl: z.string().optional(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        imageFilePath: "image_file_path",
        videoFilePath: "video_file_path",
        videoSource: "video_source",
        youtubeUrl: "youtube_url",
    });
});
exports.Schemas$PostV1FaceSwapBodyAssets = {
    in: SchemaIn$PostV1FaceSwapBodyAssets,
    out: SchemaOut$PostV1FaceSwapBodyAssets,
};
