// magic-hour/dist/types/post-v1-video-to-video-body.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schemas$PostV1VideoToVideoBody = void 0;
const post_v1_video_to_video_body_assets_1 = require("./post-v1-video-to-video-body-assets");
const post_v1_video_to_video_body_style_1 = require("./post-v1-video-to-video-body-style");
const core_1 = require("../core");
const z = __importStar(require("zod"));
/**
 * Takes network data, validates it, and transforms keys to match typescript object PostV1VideoToVideoBody
 */
const SchemaIn$PostV1VideoToVideoBody = z
    .object({
    assets: post_v1_video_to_video_body_assets_1.Schemas$PostV1VideoToVideoBodyAssets.in,
    end_seconds: z.number(),
    fps_resolution: z.enum(["FULL", "HALF"]).optional(),
    height: z.number().int(),
    name: z.string().optional(),
    start_seconds: z.number(),
    style: post_v1_video_to_video_body_style_1.Schemas$PostV1VideoToVideoBodyStyle.in,
    width: z.number().int(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        assets: "assets",
        end_seconds: "endSeconds",
        fps_resolution: "fpsResolution",
        height: "height",
        name: "name",
        start_seconds: "startSeconds",
        style: "style",
        width: "width",
    });
});
/**
 * @internal
 * Takes typescript data, validates it, and maps keys to match the expected external object External$PostV1VideoToVideoBody
 */
const SchemaOut$PostV1VideoToVideoBody = z
    .object({
    assets: post_v1_video_to_video_body_assets_1.Schemas$PostV1VideoToVideoBodyAssets.out,
    endSeconds: z.number(),
    fpsResolution: z.enum(["FULL", "HALF"]).optional(),
    height: z.number().int(),
    name: z.string().optional(),
    startSeconds: z.number(),
    style: post_v1_video_to_video_body_style_1.Schemas$PostV1VideoToVideoBodyStyle.out,
    width: z.number().int(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        assets: "assets",
        endSeconds: "end_seconds",
        fpsResolution: "fps_resolution",
        height: "height",
        name: "name",
        startSeconds: "start_seconds",
        style: "style",
        width: "width",
    });
});
exports.Schemas$PostV1VideoToVideoBody = {
    in: SchemaIn$PostV1VideoToVideoBody,
    out: SchemaOut$PostV1VideoToVideoBody,
};
