// magic-hour/dist/types/post-v1-text-to-video-body.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schemas$PostV1TextToVideoBody = void 0;
const post_v1_text_to_video_body_style_1 = require("./post-v1-text-to-video-body-style");
const core_1 = require("../core");
const z = __importStar(require("zod"));
/**
 * Takes network data, validates it, and transforms keys to match typescript object PostV1TextToVideoBody
 */
const SchemaIn$PostV1TextToVideoBody = z
    .object({
    end_seconds: z.number(),
    name: z.string().optional(),
    orientation: z.enum(["landscape", "portrait", "square"]),
    style: post_v1_text_to_video_body_style_1.Schemas$PostV1TextToVideoBodyStyle.in,
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        end_seconds: "endSeconds",
        name: "name",
        orientation: "orientation",
        style: "style",
    });
});
/**
 * @internal
 * Takes typescript data, validates it, and maps keys to match the expected external object External$PostV1TextToVideoBody
 */
const SchemaOut$PostV1TextToVideoBody = z
    .object({
    endSeconds: z.number(),
    name: z.string().optional(),
    orientation: z.enum(["landscape", "portrait", "square"]),
    style: post_v1_text_to_video_body_style_1.Schemas$PostV1TextToVideoBodyStyle.out,
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        endSeconds: "end_seconds",
        name: "name",
        orientation: "orientation",
        style: "style",
    });
});
exports.Schemas$PostV1TextToVideoBody = {
    in: SchemaIn$PostV1TextToVideoBody,
    out: SchemaOut$PostV1TextToVideoBody,
};
