// magic-hour/dist/types/post-v1-animation-body-style.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schemas$PostV1AnimationBodyStyle = void 0;
const core_1 = require("../core");
const z = __importStar(require("zod"));
/**
 * Takes network data, validates it, and transforms keys to match typescript object PostV1AnimationBodyStyle
 */
const SchemaIn$PostV1AnimationBodyStyle = z
    .object({
    art_style: z.enum([
        "3D Render",
        "90s Streets",
        "Abstract Minimalist",
        "Arcane",
        "Art Deco",
        "Bold Colored Illustration",
        "Cinematic Landscape",
        "Cinematic Miyazaki",
        "Cosmic",
        "Cubist",
        "Custom",
        "Cyberpunk",
        "Dark Graphic Illustration",
        "Dark Watercolor",
        "Directed by AI",
        "Double Exposure",
        "Faded Illustration",
        "Fantasy",
        "Futuristic Anime",
        "Impressionism",
        "Ink and Watercolor Portrait",
        "Inkpunk",
        "Intricate Abstract Lines Portrait",
        "Jackson Pollock",
        "Landscape Painting",
        "Low Poly",
        "Miniatures",
        "Minimal Cold Futurism",
        "Oil Painting",
        "Old School Comic",
        "Overgrown",
        "Painted Cityscape",
        "Painterly Illustration",
        "Photograph",
        "Pixar",
        "Pixel Art",
        "Postapocalyptic",
        "Sin City",
        "Soft Delicate Matte Portrait",
        "Spooky",
        "Studio Ghibli Film Still",
        "Synthwave",
        "Traditional Watercolor",
        "Van Gogh",
        "Vibrant Matte Illustration",
        "Vintage Japanese Anime",
        "Woodcut",
    ]),
    art_style_custom: z.string().optional(),
    camera_effect: z.enum([
        "Accelerate",
        "Aggressive Zoom In - Audio Sync",
        "Aggressive Zoom Out - Audio Sync",
        "Boost Zoom In",
        "Boost Zoom Out",
        "Bounce In And Out",
        "Bounce Out",
        "Bounce Out - Audio Sync",
        "Bounce and Spin - Audio Sync",
        "Bounce in Place",
        "Cog in the Machine",
        "Devolve - Audio Sync",
        "Directed by AI",
        "Dramatic Zoom In",
        "Dramatic Zoom Out",
        "Drift Spin",
        "Earthquake Bounce",
        "Earthquake Bounce - Audio Sync",
        "Evolve - Audio Sync",
        "Heartbeat",
        "Hesitate In",
        "Jump",
        "Pan Left",
        "Pan Right",
        "Pulse - Audio Sync",
        "Pusher",
        "Pusher - Audio Sync",
        "Quadrant",
        "Rise and Climb",
        "Road Trip",
        "Rodeo",
        "Roll In",
        "Roll In - Audio Sync",
        "Rolling Bounces",
        "Rubber Band",
        "Simple Zoom In",
        "Simple Zoom Out",
        "Slice Bounce",
        "Slideshow",
        "Speed of Light",
        "Spin Bounce",
        "Sway Out",
        "Sway Out - Audio Sync",
        "Tilt Down",
        "Tilt Up",
        "Traverse",
        "Tron",
        "Vertigo",
        "Vertigo - Audio Sync",
        "Zoom In - Audio Sync",
        "Zoom In and Spin - Audio Sync",
        "Zoom Out - Audio Sync",
    ]),
    prompt: z.string().nullable(),
    prompt_type: z.enum(["ai_choose", "custom", "use_lyrics"]),
    transition_speed: z.number().int(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        art_style: "artStyle",
        art_style_custom: "artStyleCustom",
        camera_effect: "cameraEffect",
        prompt: "prompt",
        prompt_type: "promptType",
        transition_speed: "transitionSpeed",
    });
});
/**
 * @internal
 * Takes typescript data, validates it, and maps keys to match the expected external object External$PostV1AnimationBodyStyle
 */
const SchemaOut$PostV1AnimationBodyStyle = z
    .object({
    artStyle: z.enum([
        "3D Render",
        "90s Streets",
        "Abstract Minimalist",
        "Arcane",
        "Art Deco",
        "Bold Colored Illustration",
        "Cinematic Landscape",
        "Cinematic Miyazaki",
        "Cosmic",
        "Cubist",
        "Custom",
        "Cyberpunk",
        "Dark Graphic Illustration",
        "Dark Watercolor",
        "Directed by AI",
        "Double Exposure",
        "Faded Illustration",
        "Fantasy",
        "Futuristic Anime",
        "Impressionism",
        "Ink and Watercolor Portrait",
        "Inkpunk",
        "Intricate Abstract Lines Portrait",
        "Jackson Pollock",
        "Landscape Painting",
        "Low Poly",
        "Miniatures",
        "Minimal Cold Futurism",
        "Oil Painting",
        "Old School Comic",
        "Overgrown",
        "Painted Cityscape",
        "Painterly Illustration",
        "Photograph",
        "Pixar",
        "Pixel Art",
        "Postapocalyptic",
        "Sin City",
        "Soft Delicate Matte Portrait",
        "Spooky",
        "Studio Ghibli Film Still",
        "Synthwave",
        "Traditional Watercolor",
        "Van Gogh",
        "Vibrant Matte Illustration",
        "Vintage Japanese Anime",
        "Woodcut",
    ]),
    artStyleCustom: z.string().optional(),
    cameraEffect: z.enum([
        "Accelerate",
        "Aggressive Zoom In - Audio Sync",
        "Aggressive Zoom Out - Audio Sync",
        "Boost Zoom In",
        "Boost Zoom Out",
        "Bounce In And Out",
        "Bounce Out",
        "Bounce Out - Audio Sync",
        "Bounce and Spin - Audio Sync",
        "Bounce in Place",
        "Cog in the Machine",
        "Devolve - Audio Sync",
        "Directed by AI",
        "Dramatic Zoom In",
        "Dramatic Zoom Out",
        "Drift Spin",
        "Earthquake Bounce",
        "Earthquake Bounce - Audio Sync",
        "Evolve - Audio Sync",
        "Heartbeat",
        "Hesitate In",
        "Jump",
        "Pan Left",
        "Pan Right",
        "Pulse - Audio Sync",
        "Pusher",
        "Pusher - Audio Sync",
        "Quadrant",
        "Rise and Climb",
        "Road Trip",
        "Rodeo",
        "Roll In",
        "Roll In - Audio Sync",
        "Rolling Bounces",
        "Rubber Band",
        "Simple Zoom In",
        "Simple Zoom Out",
        "Slice Bounce",
        "Slideshow",
        "Speed of Light",
        "Spin Bounce",
        "Sway Out",
        "Sway Out - Audio Sync",
        "Tilt Down",
        "Tilt Up",
        "Traverse",
        "Tron",
        "Vertigo",
        "Vertigo - Audio Sync",
        "Zoom In - Audio Sync",
        "Zoom In and Spin - Audio Sync",
        "Zoom Out - Audio Sync",
    ]),
    prompt: z.string().nullable(),
    promptType: z.enum(["ai_choose", "custom", "use_lyrics"]),
    transitionSpeed: z.number().int(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        artStyle: "art_style",
        artStyleCustom: "art_style_custom",
        cameraEffect: "camera_effect",
        prompt: "prompt",
        promptType: "prompt_type",
        transitionSpeed: "transition_speed",
    });
});
exports.Schemas$PostV1AnimationBodyStyle = {
    in: SchemaIn$PostV1AnimationBodyStyle,
    out: SchemaOut$PostV1AnimationBodyStyle,
};
