// magic-hour/dist/types/post-v1-ai-photo-editor-body-style.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schemas$PostV1AiPhotoEditorBodyStyle = void 0;
const core_1 = require("../core");
const z = __importStar(require("zod"));
/**
 * Takes network data, validates it, and transforms keys to match typescript object PostV1AiPhotoEditorBodyStyle
 */
const SchemaIn$PostV1AiPhotoEditorBodyStyle = z
    .object({
    image_description: z.string(),
    likeness_strength: z.number(),
    negative_prompt: z.string().optional(),
    prompt: z.string(),
    prompt_strength: z.number(),
    steps: z.number().int().optional(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        image_description: "imageDescription",
        likeness_strength: "likenessStrength",
        negative_prompt: "negativePrompt",
        prompt: "prompt",
        prompt_strength: "promptStrength",
        steps: "steps",
    });
});
/**
 * @internal
 * Takes typescript data, validates it, and maps keys to match the expected external object External$PostV1AiPhotoEditorBodyStyle
 */
const SchemaOut$PostV1AiPhotoEditorBodyStyle = z
    .object({
    imageDescription: z.string(),
    likenessStrength: z.number(),
    negativePrompt: z.string().optional(),
    prompt: z.string(),
    promptStrength: z.number(),
    steps: z.number().int().optional(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        imageDescription: "image_description",
        likenessStrength: "likeness_strength",
        negativePrompt: "negative_prompt",
        prompt: "prompt",
        promptStrength: "prompt_strength",
        steps: "steps",
    });
});
exports.Schemas$PostV1AiPhotoEditorBodyStyle = {
    in: SchemaIn$PostV1AiPhotoEditorBodyStyle,
    out: SchemaOut$PostV1AiPhotoEditorBodyStyle,
};
