// magic-hour/dist/types/post-v1-face-swap-response.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schemas$PostV1FaceSwapResponse = void 0;
const core_1 = require("../core");
const z = __importStar(require("zod"));
/**
 * Takes network data, validates it, and transforms keys to match typescript object PostV1FaceSwapResponse
 */
const SchemaIn$PostV1FaceSwapResponse = z
    .object({
    estimated_frame_cost: z.number().int(),
    id: z.string(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        estimated_frame_cost: "estimatedFrameCost",
        id: "id",
    });
});
/**
 * @internal
 * Takes typescript data, validates it, and maps keys to match the expected external object External$PostV1FaceSwapResponse
 */
const SchemaOut$PostV1FaceSwapResponse = z
    .object({
    estimatedFrameCost: z.number().int(),
    id: z.string(),
})
    .transform((obj) => {
    return (0, core_1.zodTransform)(obj, {
        estimatedFrameCost: "estimated_frame_cost",
        id: "id",
    });
});
exports.Schemas$PostV1FaceSwapResponse = {
    in: SchemaIn$PostV1FaceSwapResponse,
    out: SchemaOut$PostV1FaceSwapResponse,
};
