// magic-hour/dist/resources/v1/text-to-video/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextToVideoClient = void 0;
const core_1 = require("../../../core");
const post_v1_text_to_video_body_1 = require("../../../types/post-v1-text-to-video-body");
const post_v1_text_to_video_response_1 = require("../../../types/post-v1-text-to-video-response");
class TextToVideoClient extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
    }
    /**
     * Text-to-Video
     *
     * Create a Text To Video video. The estimated frame cost is calculated using 30 FPS. This amount is deducted from your account balance when a video is queued. Once the video is complete, the cost will be updated based on the actual number of frames rendered.
     *
     * Get more information about this mode at our [product page](/products/text-to-video).
     *
     *
     * POST /v1/text-to-video
     */
    create(request, opts) {
        return this._client.makeRequest({
            method: "post",
            path: "/v1/text-to-video",
            auth: ["bearerAuth"],
            contentType: "application/json",
            body: post_v1_text_to_video_body_1.Schemas$PostV1TextToVideoBody.out.parse(request),
            responseSchema: post_v1_text_to_video_response_1.Schemas$PostV1TextToVideoResponse.in,
            opts,
        });
    }
}
exports.TextToVideoClient = TextToVideoClient;
