// magic-hour/dist/resources/v1/ai-image-generator/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiImageGeneratorClient = void 0;
const core_1 = require("../../../core");
const post_v1_ai_image_generator_body_1 = require("../../../types/post-v1-ai-image-generator-body");
const post_v1_ai_image_generator_response_1 = require("../../../types/post-v1-ai-image-generator-response");
class AiImageGeneratorClient extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
    }
    /**
     * AI Images
     *
     * Create an AI image. Each image costs 5 frames.
     *
     * POST /v1/ai-image-generator
     */
    create(request, opts) {
        return this._client.makeRequest({
            method: "post",
            path: "/v1/ai-image-generator",
            auth: ["bearerAuth"],
            contentType: "application/json",
            body: post_v1_ai_image_generator_body_1.Schemas$PostV1AiImageGeneratorBody.out.parse(request),
            responseSchema: post_v1_ai_image_generator_response_1.Schemas$PostV1AiImageGeneratorResponse.in,
            opts,
        });
    }
}
exports.AiImageGeneratorClient = AiImageGeneratorClient;
