// magic-hour/dist/resources/v1/image-background-remover/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageBackgroundRemoverClient = void 0;
const core_1 = require("../../../core");
const post_v1_image_background_remover_body_1 = require("../../../types/post-v1-image-background-remover-body");
const post_v1_image_background_remover_response_1 = require("../../../types/post-v1-image-background-remover-response");
class ImageBackgroundRemoverClient extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
    }
    /**
     * Image Background Remover
     *
     * Remove background from image. Each image costs 5 frames.
     *
     * POST /v1/image-background-remover
     */
    create(request, opts) {
        return this._client.makeRequest({
            method: "post",
            path: "/v1/image-background-remover",
            auth: ["bearerAuth"],
            contentType: "application/json",
            body: post_v1_image_background_remover_body_1.Schemas$PostV1ImageBackgroundRemoverBody.out.parse(request),
            responseSchema: post_v1_image_background_remover_response_1.Schemas$PostV1ImageBackgroundRemoverResponse.in,
            opts,
        });
    }
}
exports.ImageBackgroundRemoverClient = ImageBackgroundRemoverClient;
