// magic-hour/dist/resources/v1/face-swap-photo/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FaceSwapPhotoClient = void 0;
const core_1 = require("../../../core");
const post_v1_face_swap_photo_body_1 = require("../../../types/post-v1-face-swap-photo-body");
const post_v1_face_swap_photo_response_1 = require("../../../types/post-v1-face-swap-photo-response");
class FaceSwapPhotoClient extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
    }
    /**
     * Face Swap Photo
     *
     * Create a face swap photo. Each photo costs 5 frames. The height/width of the output image depends on your subscription. Please refer to our [pricing](/pricing) page for more details
     *
     * POST /v1/face-swap-photo
     */
    create(request, opts) {
        return this._client.makeRequest({
            method: "post",
            path: "/v1/face-swap-photo",
            auth: ["bearerAuth"],
            contentType: "application/json",
            body: post_v1_face_swap_photo_body_1.Schemas$PostV1FaceSwapPhotoBody.out.parse(request),
            responseSchema: post_v1_face_swap_photo_response_1.Schemas$PostV1FaceSwapPhotoResponse.in,
            opts,
        });
    }
}
exports.FaceSwapPhotoClient = FaceSwapPhotoClient;
