// magic-hour/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.types = exports.resources = exports.Environment = exports.isUploadFile = exports.BinaryResponse = exports.ApiError = exports.default = void 0;
var client_1 = require("./client");
Object.defineProperty(exports, "default", { enumerable: true, get: function () { return client_1.Client; } });
var core_1 = require("./core");
Object.defineProperty(exports, "ApiError", { enumerable: true, get: function () { return core_1.ApiError; } });
Object.defineProperty(exports, "BinaryResponse", { enumerable: true, get: function () { return core_1.BinaryResponse; } });
Object.defineProperty(exports, "isUploadFile", { enumerable: true, get: function () { return core_1.isUploadFile; } });
var environment_1 = require("./environment");
Object.defineProperty(exports, "Environment", { enumerable: true, get: function () { return environment_1.Environment; } });
exports.resources = __importStar(require("./resources"));
exports.types = __importStar(require("./types"));
