// magic-hour/dist/resources/v1/ai-image-upscaler/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiImageUpscalerClient = void 0;
const core_1 = require("../../../core");
const post_v1_ai_image_upscaler_body_1 = require("../../../types/post-v1-ai-image-upscaler-body");
const post_v1_ai_image_upscaler_response_1 = require("../../../types/post-v1-ai-image-upscaler-response");
class AiImageUpscalerClient extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
    }
    /**
     * AI Image Upscaler
     *
     * Upscale your image using AI. Each 2x upscale costs 50 frames, and 4x upscale costs 200 frames.
     *
     * POST /v1/ai-image-upscaler
     */
    create(request, opts) {
        return this._client.makeRequest({
            method: "post",
            path: "/v1/ai-image-upscaler",
            auth: ["bearerAuth"],
            contentType: "application/json",
            body: post_v1_ai_image_upscaler_body_1.Schemas$PostV1AiImageUpscalerBody.out.parse(request),
            responseSchema: post_v1_ai_image_upscaler_response_1.Schemas$PostV1AiImageUpscalerResponse.in,
            opts,
        });
    }
}
exports.AiImageUpscalerClient = AiImageUpscalerClient;
