// magic-hour/dist/resources/v1/ai-qr-code-generator/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiQrCodeGeneratorClient = void 0;
const core_1 = require("../../../core");
const post_v1_ai_qr_code_generator_body_1 = require("../../../types/post-v1-ai-qr-code-generator-body");
const post_v1_ai_qr_code_generator_response_1 = require("../../../types/post-v1-ai-qr-code-generator-response");
class AiQrCodeGeneratorClient extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
    }
    /**
     * AI QR Code
     *
     * Create an AI QR code. Each QR code costs 20 frames.
     *
     * POST /v1/ai-qr-code-generator
     */
    create(request, opts) {
        return this._client.makeRequest({
            method: "post",
            path: "/v1/ai-qr-code-generator",
            auth: ["bearerAuth"],
            contentType: "application/json",
            body: post_v1_ai_qr_code_generator_body_1.Schemas$PostV1AiQrCodeGeneratorBody.out.parse(request),
            responseSchema: post_v1_ai_qr_code_generator_response_1.Schemas$PostV1AiQrCodeGeneratorResponse.in,
            opts,
        });
    }
}
exports.AiQrCodeGeneratorClient = AiQrCodeGeneratorClient;
