// magic-hour/dist/resources/v1/ai-clothes-changer/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiClothesChangerClient = void 0;
const core_1 = require("../../../core");
const post_v1_ai_clothes_changer_body_1 = require("../../../types/post-v1-ai-clothes-changer-body");
const post_v1_ai_clothes_changer_response_1 = require("../../../types/post-v1-ai-clothes-changer-response");
class AiClothesChangerClient extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
    }
    /**
     * AI Clothes Changer
     *
     * Change outfits in photos in seconds with just a photo reference. Each photo costs 25 frames.
     *
     * POST /v1/ai-clothes-changer
     */
    create(request, opts) {
        return this._client.makeRequest({
            method: "post",
            path: "/v1/ai-clothes-changer",
            auth: ["bearerAuth"],
            contentType: "application/json",
            body: post_v1_ai_clothes_changer_body_1.Schemas$PostV1AiClothesChangerBody.out.parse(request),
            responseSchema: post_v1_ai_clothes_changer_response_1.Schemas$PostV1AiClothesChangerResponse.in,
            opts,
        });
    }
}
exports.AiClothesChangerClient = AiClothesChangerClient;
