// magic-hour/dist/resources/v1/ai-headshot-generator/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiHeadshotGeneratorClient = void 0;
const core_1 = require("../../../core");
const post_v1_ai_headshot_generator_body_1 = require("../../../types/post-v1-ai-headshot-generator-body");
const post_v1_ai_headshot_generator_response_1 = require("../../../types/post-v1-ai-headshot-generator-response");
class AiHeadshotGeneratorClient extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
    }
    /**
     * AI Headshots
     *
     * Create an AI headshot. Each headshot costs 50 frames.
     *
     * POST /v1/ai-headshot-generator
     */
    create(request, opts) {
        return this._client.makeRequest({
            method: "post",
            path: "/v1/ai-headshot-generator",
            auth: ["bearerAuth"],
            contentType: "application/json",
            body: post_v1_ai_headshot_generator_body_1.Schemas$PostV1AiHeadshotGeneratorBody.out.parse(request),
            responseSchema: post_v1_ai_headshot_generator_response_1.Schemas$PostV1AiHeadshotGeneratorResponse.in,
            opts,
        });
    }
}
exports.AiHeadshotGeneratorClient = AiHeadshotGeneratorClient;
