// magic-hour/dist/resources/v1/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.videoToVideo = exports.videoProjects = exports.textToVideo = exports.V1Client = exports.lipSync = exports.imageToVideo = exports.imageProjects = exports.imageBackgroundRemover = exports.files = exports.faceSwapPhoto = exports.faceSwap = exports.animation = exports.aiQrCodeGenerator = exports.aiPhotoEditor = exports.aiImageUpscaler = exports.aiImageGenerator = exports.aiHeadshotGenerator = exports.aiClothesChanger = void 0;
exports.aiClothesChanger = __importStar(require("./ai-clothes-changer"));
exports.aiHeadshotGenerator = __importStar(require("./ai-headshot-generator"));
exports.aiImageGenerator = __importStar(require("./ai-image-generator"));
exports.aiImageUpscaler = __importStar(require("./ai-image-upscaler"));
exports.aiPhotoEditor = __importStar(require("./ai-photo-editor"));
exports.aiQrCodeGenerator = __importStar(require("./ai-qr-code-generator"));
exports.animation = __importStar(require("./animation"));
exports.faceSwap = __importStar(require("./face-swap"));
exports.faceSwapPhoto = __importStar(require("./face-swap-photo"));
exports.files = __importStar(require("./files"));
exports.imageBackgroundRemover = __importStar(require("./image-background-remover"));
exports.imageProjects = __importStar(require("./image-projects"));
exports.imageToVideo = __importStar(require("./image-to-video"));
exports.lipSync = __importStar(require("./lip-sync"));
var resource_client_1 = require("./resource-client");
Object.defineProperty(exports, "V1Client", { enumerable: true, get: function () { return resource_client_1.V1Client; } });
exports.textToVideo = __importStar(require("./text-to-video"));
exports.videoProjects = __importStar(require("./video-projects"));
exports.videoToVideo = __importStar(require("./video-to-video"));
