// magic-hour/dist/resources/v1/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.V1Client = void 0;
const core_1 = require("../../core");
const ai_clothes_changer_1 = require("../../resources/v1/ai-clothes-changer");
const ai_headshot_generator_1 = require("../../resources/v1/ai-headshot-generator");
const ai_image_generator_1 = require("../../resources/v1/ai-image-generator");
const ai_image_upscaler_1 = require("../../resources/v1/ai-image-upscaler");
const ai_photo_editor_1 = require("../../resources/v1/ai-photo-editor");
const ai_qr_code_generator_1 = require("../../resources/v1/ai-qr-code-generator");
const animation_1 = require("../../resources/v1/animation");
const face_swap_1 = require("../../resources/v1/face-swap");
const face_swap_photo_1 = require("../../resources/v1/face-swap-photo");
const files_1 = require("../../resources/v1/files");
const image_background_remover_1 = require("../../resources/v1/image-background-remover");
const image_projects_1 = require("../../resources/v1/image-projects");
const image_to_video_1 = require("../../resources/v1/image-to-video");
const lip_sync_1 = require("../../resources/v1/lip-sync");
const text_to_video_1 = require("../../resources/v1/text-to-video");
const video_projects_1 = require("../../resources/v1/video-projects");
const video_to_video_1 = require("../../resources/v1/video-to-video");
class V1Client extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
        this.imageProjects = new image_projects_1.ImageProjectsClient(this._client);
        this.videoProjects = new video_projects_1.VideoProjectsClient(this._client);
        this.aiClothesChanger = new ai_clothes_changer_1.AiClothesChangerClient(this._client);
        this.aiHeadshotGenerator = new ai_headshot_generator_1.AiHeadshotGeneratorClient(this._client);
        this.aiImageGenerator = new ai_image_generator_1.AiImageGeneratorClient(this._client);
        this.aiImageUpscaler = new ai_image_upscaler_1.AiImageUpscalerClient(this._client);
        this.aiPhotoEditor = new ai_photo_editor_1.AiPhotoEditorClient(this._client);
        this.aiQrCodeGenerator = new ai_qr_code_generator_1.AiQrCodeGeneratorClient(this._client);
        this.animation = new animation_1.AnimationClient(this._client);
        this.faceSwap = new face_swap_1.FaceSwapClient(this._client);
        this.faceSwapPhoto = new face_swap_photo_1.FaceSwapPhotoClient(this._client);
        this.files = new files_1.FilesClient(this._client);
        this.imageBackgroundRemover = new image_background_remover_1.ImageBackgroundRemoverClient(this._client);
        this.imageToVideo = new image_to_video_1.ImageToVideoClient(this._client);
        this.lipSync = new lip_sync_1.LipSyncClient(this._client);
        this.textToVideo = new text_to_video_1.TextToVideoClient(this._client);
        this.videoToVideo = new video_to_video_1.VideoToVideoClient(this._client);
    }
}
exports.V1Client = V1Client;
