// magic-hour/dist/resources/v1/video-projects/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VideoProjectsClient = void 0;
const core_1 = require("../../../core");
const get_v1_video_projects_id_response_1 = require("../../../types/get-v1-video-projects-id-response");
class VideoProjectsClient extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
    }
    /**
     * Get video details
     *
     * Get the details of a video project. The `downloads` field will be empty unless the video was successfully rendered.
     *
     * The video can be one of the following status
     * - `draft` - not currently used
     * - `queued` - the job is queued and waiting for a GPU
     * - `rendering` - the generation is in progress
     * - `complete` - the video is successful created
     * - `error` - an error occurred during rendering
     * - `canceled` - video render is canceled by the user
     *
     *
     * GET /v1/video-projects/{id}
     */
    get(request, opts) {
        return this._client.makeRequest({
            method: "get",
            path: `/v1/video-projects/${request.id}`,
            auth: ["bearerAuth"],
            responseSchema: get_v1_video_projects_id_response_1.Schemas$GetV1VideoProjectsIdResponse.in,
            opts,
        });
    }
    /**
     * Delete video
     *
     * Permanently delete the rendered video. This action is not reversible, please be sure before deleting.
     *
     * DELETE /v1/video-projects/{id}
     */
    delete(request, opts) {
        return this._client.makeRequest({
            method: "delete",
            path: `/v1/video-projects/${request.id}`,
            auth: ["bearerAuth"],
            responseRaw: true,
            opts,
        });
    }
}
exports.VideoProjectsClient = VideoProjectsClient;
