// magic-hour/dist/resources/v1/animation/resource-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnimationClient = void 0;
const core_1 = require("../../../core");
const post_v1_animation_body_1 = require("../../../types/post-v1-animation-body");
const post_v1_animation_response_1 = require("../../../types/post-v1-animation-response");
class AnimationClient extends core_1.CoreResourceClient {
    constructor(client) {
        super(client);
    }
    /**
     * Animation
     *
     * Create a Animation video. The estimated frame cost is calculated based on the `fps` and `end_seconds` input.
     *
     * POST /v1/animation
     */
    create(request, opts) {
        return this._client.makeRequest({
            method: "post",
            path: "/v1/animation",
            auth: ["bearerAuth"],
            contentType: "application/json",
            body: post_v1_animation_body_1.Schemas$PostV1AnimationBody.out.parse(request),
            responseSchema: post_v1_animation_response_1.Schemas$PostV1AnimationResponse.in,
            opts,
        });
    }
}
exports.AnimationClient = AnimationClient;
